code/modules/mob/mob.dm

```diff
@@ -331,15 +331,66 @@
 	if(!I)
 		return FALSE
 
-	if(istype(I, /obj/item/storage/belt/gun))
+	//the slot define specific early returns
+	if (slot == SLOT_IN_HOLSTER && ( !( istype(I, /obj/item/storage/holster) || istype(I, /obj/item/weapon) || istype(I, /obj/item/storage/belt/gun) ) ) )
+		return FALSE
+	if (slot == SLOT_IN_S_HOLSTER && ( !( istype(I, /obj/item/storage/holster) || istype(I, /obj/item/weapon) || istype(I, /obj/item/storage/belt/gun) || istype(I, /obj/item/storage/belt/knifepouch) ) ) )
+		return FALSE
+	if (slot == SLOT_IN_B_HOLSTER && ( !( istype(I, /obj/item/storage/holster) || istype(I, /obj/item/weapon) ) ) )
+		return FALSE
+	if (slot == SLOT_IN_ACCESSORY && (!istype(I, /obj/item/clothing/under ) ) ) //note to self, draw code has a further istype for the attachment, does this also need an early return if there is no attachment, or it's fine? probs fine
+		return FALSE
+	if (slot == SLOT_IN_L_POUCH && ( !( istype(I, /obj/item/storage/holster) || istype(I, /obj/item/weapon) || istype(I, /obj/item/storage/pouch/pistol) ) ) ) //if pistol pouches get merged into holsters, this will need to be updated
+		return FALSE
+	if (slot == SLOT_IN_R_POUCH && ( !( istype(I, /obj/item/storage/holster) || istype(I, /obj/item/weapon) || istype(I, /obj/item/storage/pouch/pistol) ) ) )
+		return FALSE
+
+	//belt holsters
```

Comment:
this massive if chaining is ugly
also please dont do (  ( ( ( ) ) ) ) do (((())) instead

Reply:
Will do.